File: tests/unit/fixtures/lint_http/allowed/R/lib/http.R
# The R chokepoint. Every idiom the linter knows appears here on purpose.
library(httr2)
library(curl)
absump_http_get <- function(u, dest) {
  resp <- httr2::request(u) |> req_perform()
  download.file(u, destfile = dest)
  httr::GET(u)
  txt <- readLines(url(u))
  out <- system2("curl", c("-s", u), stdout = TRUE)
  invisible(list(resp, txt, out))
}

File: tests/unit/fixtures/lint_http/allowed/src/absump/http.py
# ruff: noqa
# The Python chokepoint. Every idiom the linter knows appears here on purpose.
import http.client
import socket
import subprocess
import urllib.request
from urllib.request import urlopen

import aiohttp
import httpx
import requests
from httpx import get
from requests import get as g


def fetch(url: str) -> bytes:
    client = httpx.Client()
    httpx.post(url, json={})
    requests.get(url)
    http.client.HTTPSConnection("statsapi.mlb.com")
    socket.create_connection(("statsapi.mlb.com", 443))
    subprocess.run(["curl", "-s", url], check=True)
    return client.get(url).content
